proposals/NNNN-atf_security_proposal.md

```diff
@@ -0,0 +1,135 @@
+
+Should be added new ini file option: `SecurityProtocol`\
+Should be added new command line option: `security_protocol`\
+In case if this option is missed, use TLS security by default.
+
+### Detailed design
+
+#### New component: 
+Will be added new component : *SecurityManager* 
+
+Responsibility of SecurityManager :
+ - Manage digital certificates;
+ - Manage TLS or DTLS protocols; 
+ - Handle Secure Sockets Layer (SSL) context;
+ - Provide interface for encryption\decryption;
+ - Be able to perform a handshake.
+
+#### New APIs: 
+
+##### connection.StartSecureSession:
+###### Description
+ Start secure session :
+  1. Send StartSession with `encrypted flag = true`;
+  2. Perform TLS/DTLS handshake; 
+  3. Expect StartSessionAck.
+  
+  This is a blocking call. Will block current execution until session won't be established or failed.
+###### Arguments:
+  - protocol (if missed, used one from console or ini file options).
+###### Return value:
+  - session object.
+  
+##### session.SendEncryptedRPC:
+###### Description
+ Send encrypted RPC:
+  1. Encrypts payload and binary data;
+  2. Send RPC to SDL;
+###### Arguments:
+  - function name - stringified RPC name;  
+  - arguments  - lua table with arguments of RPC (payload);
+  - file name - path to file with binary data.
+###### Return value:
+  - correlation ID of sent request.
+
+##### session.ExpectEncryptedResponse:
+###### Description
+ Add expectation to encrypted response with specific correlation_id.
+###### Arguments :
+  - correlation_id - correlation ID;
+  - data  - expected payload (decrypted).
+###### Return value :
+  - expectation.
+  
+##### session.ExpectEncryptedNotification:
+###### Description
+ Add expectation to encrypted notification.
+###### Arguments:
+  - function_name - expected notificaiton name
+  - data - expected payload (decrypted).
+###### Return value:
+  - expectation.
+  
+##### session.SendPacket:
+###### Description
+Send Raw Packet for testing wrong TLS handshake.
+###### Arguments:
+  - data - bytes to send.
+###### Return value :
+  N\A
+  
+##### session.ExpectPacket :
+###### Description
+Expect custom packet for checking TLS Handshake.
+###### Arguments :
+  - data - bytes that ATF expects.
+###### Return value :
+  - expectation.
+
+## Potential downsides
+
+- session.ExpectPacket - very common interface, it might need more deep analysis of input data. Also present possibility to reduce efficiency of the whole ATF system. This interface also allows tester to check any incoming data from SDL.
+
+- session.SendPacket - very low level interface, it might require a lot of additional complicated logic for constructing bytes to send in script. This interface also allows tester to send any data.
+
+## Impact on existing code
+
+Should be impacted only ATF code.
+If during implementation will be found some blocker issues in SDL, they should be fixed.

```

Comment:
"If blocker issues in SDL Core are found during implementation, the should be fixed."